enterprise/frontend/src/metabase-enterprise/rich_text_editing/tiptap/extensions/HandleEditorDrop/HandleEditorDrop.tsx

```diff
@@ -0,0 +1,827 @@
+import { Extension } from "@tiptap/core";
+import {
+  Fragment,
+  type Node,
+  type NodeType as PMNodeType,
+  Slice,
+} from "@tiptap/pm/model";
+import { NodeSelection, Plugin, PluginKey } from "@tiptap/pm/state";
+import type { EditorView as PMEditorView } from "@tiptap/pm/view";
+import type { NodeViewProps } from "@tiptap/react";
+
+import {
+  RESIZE_NODE_DEFAULT_HEIGHT,
+  RESIZE_NODE_MIN_HEIGHT,
+} from "metabase-enterprise/rich_text_editing/tiptap/extensions/ResizeNode/ResizeNode";
+
+import {
+  type DroppedCardEmbedNodeData,
+  extractCardEmbed,
+  getCardEmbedDropSide,
+  getDroppedCardEmbedNodeData,
+  getTargetCardEmbedNode,
+} from "./utils";
+
+declare module "prosemirror-view" {
+  // This adds a new configuration option to the NodeConfig
+  class EditorView {
+    draggingNode?: NodeViewProps["node"] | null;
+  }
+}
+
+export const HandleEditorDrop = Extension.create({
+  name: "handleEditorDrop",
+  addProseMirrorPlugins() {
+    return [
+      new Plugin({
+        key: new PluginKey("metabaseTiptapDrop"),
+        props: {
+          // eslint-disable-next-line complexity
```

Comment:
We should break this up a bit so that we don't need to disable this line. For instance, we could pass dropping into a flex container into it's own function